src/rpc/misc.cpp

```diff
@@ -28,6 +29,7 @@ static UniValue validateaddress(const JSONRPCRequest& request)
                 "\nReturn information about the given bitcoin address.\n",
                 {
                     {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The bitcoin address to validate"},
+                    {"address_type", RPCArg::Type::STR, RPCArg::Optional::OMITTED_NAMED_ARG, "The address type provided, used to detect errors. Options are \"legacy\", \"p2sh-segwit\", and \"bech32\"."},
```

Comment:
Why can't this be detected from the input?

Reply:
Because if we assume it has errors in it, then we can't rely on checks like whether all the characters are bech32 or base58 or whatever. But I like Sjors' suggestion of counting the characters to find the most likely. I'll try that when I rebase this.

Reply:
FWIW, for Knots I am just checking for the Bech32HRP to match

Reply:
That sounds simple enough.

Reply:
https://github.com/bitcoinknots/bitcoin/commit/65b45fa8f852a6d375061929e6c943c2afbbe681